keps/sig-api-machinery/2876-crd-validation-expression-language/README.md

```diff
@@ -228,6 +225,23 @@ kind: CustomResourceDefinition
               type: integer
 ```
 
+Example Validation Rules:
+
+| Rule                                                                             | Purpose                                                                           |
+| ----------------                                                                 | ------------                                                                      |
+| `self.minReplicas <= self.replicas <= self.maxReplicas`                          | Validate that the three fields defining replicas are ordered appropriately        |
+| `'Available' in self.stateCounts`                                                | Validate that an entry with the 'Available' key exists in a map                   |
+| `(size(self.list1) == 0) != (size(self.list2) == 0)`                             | Validate that one of two lists is non-empty, but not both                         |
+| `!('MY_ENV' in self.envars) || self['MY_ENV'].matches('^[a-zA-Z]*$')`            | Validate the value of a map for a specific key, if it is in the map               |
```

Comment:
env vars are a list, not a map.

Reply:
Ahh, right. Someone reminded me a while ago that this it the use case that is most used to highlight why listMap order matters.

I've put the example in as a list. I kept the map example but changed it to not use envars.